File: objc/Project11/Project11 WatchKit Extension/InterfaceController.m
//
//  InterfaceController.m
//  Project11 WatchKit Extension
//
//  Created by Jinwoo Kim on 2/1/21.
//

#import "InterfaceController.h"
#import "GameScene.h"

@implementation InterfaceController

- (void)awakeWithContext:(id)context {
    [super awakeWithContext:context];
    [self startGame:self];
}

- (void)didAppear {
    [super didAppear];
    [self.crownSequencer focus];
}

- (IBAction)startGame:(id)sender {
    if ((self.gameScene) && (self.gameScene.isPlayerAlive)) return;
    
    self.gameScene = [GameScene sceneWithSize:CGSizeMake(self.contentFrame.size.width, self.contentFrame.size.height)];
    self.gameScene.parentInterfaceController = self;
    self.gameScene.anchorPoint = CGPointMake(0.5, 0.5);
    
    SKTransition *transition = [SKTransition doorwayWithDuration:1];
    self.crownSequencer.delegate = self.gameScene;
    [self.gameInterface presentScene:self.gameScene transition:transition];
}

@end

File: swift/Project11/Project11 WatchKit Extension/InterfaceController.swift
//
//  InterfaceController.swift
//  Project11 WatchKit Extension
//
//  Created by Jinwoo Kim on 1/31/21.
//

import WatchKit
import Foundation
import SpriteKit

class InterfaceController: WKInterfaceController {
    @IBOutlet weak var gameInterface: WKInterfaceSKScene!
    var gameScene: GameScene!
    
    override func awake(withContext context: Any?) {
        super.awake(withContext: context)
        startGame(self)
    }
    
    override func didAppear() {
        super.didAppear()
        crownSequencer.focus()
    }

    @IBAction func startGame(_ sender: Any) {
        if gameScene != nil {
            guard gameScene.isPlayerAlive == false else { return }
        }
        
        gameScene = GameScene(size: CGSize(width: contentFrame.width, height: contentFrame.height))
        gameScene.parentInterfaceController = self
        gameScene.anchorPoint = CGPoint(x: 0.5, y: 0.5)
        
        let transition = SKTransition.doorway(withDuration: 1)
        crownSequencer.delegate = gameScene
        gameInterface.presentScene(gameScene, transition: transition)
    }
}
